transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/internal/util/Util.java

```diff
@@ -111,14 +109,19 @@ public static boolean draggingFromValidTarget(TransformationManager manager) {
     }
 
     public static void ensureSourceFolderExists(IJavaProject javaProject,
-                                                String folder,
-                                                IProgressMonitor monitor) throws JavaModelException {
+                                                String folderName,
+                                                IProgressMonitor monitor) throws Exception {
         boolean exists = false;
         IClasspathEntry[] entries = javaProject.getRawClasspath();
-        IPath path = javaProject.getPath().append(folder);
+        IPath path = javaProject.getPath().append(folderName);
         for (IClasspathEntry entry : entries) {
             if (entry.getEntryKind() == IClasspathEntry.CPE_SOURCE && entry.getPath().equals(path)) {
                 exists = true;
+                File folder = javaProject.getResource().getLocation().append(folderName).toFile();
+                if (!folder.exists()) {
+                    if (!folder.mkdirs()) throw new Exception(Messages.bind(Messages.Util_UnableToCreateSourceFolder, folder));
```

Comment:
IOException is more explicit than a common Exception.
Having specific exception thrown will help us to wrap them and provide specific feedback later on. using generic Exception we will have to dig through the whole code which exceptions can really be thrown.
I would prefer to do it from the start (but I won't stop the merge of the PR for that)

For multiple statements on a single line, using them is religious but having them on a separated line is scientific. It greatly improves readability and avoids ambiguity. All software quality tools that I met enforced this way of doing it. (again i won't block the PR for this point)